### ISSUE TITLE
checksum failes on msi installer

### ISSUE THREAD

**Issue by Renelast** [puppetlabs/puppetlabs-puppet_agent#195]
The checksum sha256lite fails when downloading the installer msi file on a windows client.
``
change from absent to file failed: File written to disk did not match checksum; discarding changes ({sha256lite}5463b3053faecf798f463c5b019a00a7542105829d6b16f6882df670eb81679a vs {sha256lite}01009e50abac71f031a3204235a9864fe3be292d602cef645fe1a47ea8672dbd)
Source:	/Stage[main]/Puppet_agent::Prepare::Package/File[C:\ProgramData\Puppetlabs\packages\puppet-agent-x64.msi]/ensureFile:	/etc/puppetlabs/code/environments/test/modules/puppet_agent/manifests/prepare/package.ppLine:	4
``
In our case the puppet client version is 3.8.1 and the server is running pe2016.4.2
I have downloaded the file manually on a test windows client and was able to verify the sha256lite checksum is indeed different from the one I get on the server.

**Comment by MosesMendoza** [comment]
thanks for filing this @Renelast - we're looking into it

**Comment by MosesMendoza** [comment]
@Renelast just to confirm I understand - 
you have a 2016.4.2 server, and you are upgrading a puppet client from version puppet 3.8.1 to puppet-agent 1.7.1 (shipped with 2016.4.2) ?

**Comment by Renelast** [comment]
@MosesMendoza That is correct. The clients are being migrated from an old PE 3.8.1 server to this a new pe 2016.4.2 server.

**Comment by Renelast** [comment]
@MosesMendoza as a workaround I changed the checksum type to md5 which seems to work.
Client Os is windows 2012 R2 by the way..

**Comment by MosesMendoza** [comment]
@Renelast also - what version of the puppet agent module are you running?

**Comment by MosesMendoza** [comment]
I can reproduce this when using this module from HEAD, but not from version 1.3.1 (latest release).

With 1.3.1:
```
PS C:\Users\Administrator> puppet agent -t
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for win-pp9ct9knt05.localdomain
Info: Applying configuration version '1481054153'
Notice: /Stage[main]/Puppet_agent::Prepare::Puppet_config/Ini_setting[main/stringify_facts]/ensure: removed
Notice: /Stage[main]/Puppet_agent::Prepare::Puppet_config/Ini_setting[main/pluginsync]/ensure: removed
Notice: /Stage[main]/Puppet_agent::Prepare::Package/File[C:\ProgramData\Puppetlabs\packages]/ensure: created
Notice: /Stage[main]/Puppet_agent::Prepare::Package/File[C:\ProgramData\Puppetlabs\packages\puppet-agent-x64.msi]/ensure
: defined content as '{md5}4a894b9780029acd281b56438a00faa7'
Notice: /Stage[main]/Puppet_agent::Windows::Install/File[C:\Users\ADMINI~1\AppData\Local\Temp\install_puppet.bat]/ensure
: defined content as '{md5}0ae762dd31a96592125ff9645dddc4c3'
Notice: /Stage[main]/Puppet_agent::Windows::Install/Exec[install_puppet.bat]/returns: executed successfully
Notice: /Stage[main]/Puppet_agent::Windows::Install/Exec[fix inheritable SYSTEM perms]/returns: executed successfully
```

from HEAD
```
Notice: /Stage[main]/Puppet_agent::Prepare::Puppet_config/Ini_setting[main/stringify_facts]/ensure: removed
Notice: /Stage[main]/Puppet_agent::Prepare::Puppet_config/Ini_setting[main/pluginsync]/ensure: removed
Notice: /Stage[main]/Puppet_agent::Prepare::Package/File[C:\ProgramData\Puppetlabs\packages]/ensure: created
Error: File written to disk did not match checksum; discarding changes ({sha256lite}5463b3053faecf798f463c5b019a00a75421
05829d6b16f6882df670eb81679a vs {sha256lite}01009e50abac71f031a3204235a9864fe3be292d602cef645fe1a47ea8672dbd)
Error: /Stage[main]/Puppet_agent::Prepare::Package/File[C:\ProgramData\Puppetlabs\packages\puppet-agent-x64.msi]/ensure:
 change from absent to file failed: File written to disk did not match checksum; discarding changes ({sha256lite}5463b30
53faecf798f463c5b019a00a7542105829d6b16f6882df670eb81679a vs {sha256lite}01009e50abac71f031a3204235a9864fe3be292d602cef6
45fe1a47ea8672dbd)
Notice: /Stage[main]/Puppet_agent::Windows::Install/File[C:\Users\ADMINI~1\AppData\Local\Temp\install_puppet.bat]: Depen
dency File[C:\ProgramData\Puppetlabs\packages\puppet-agent-x64.msi] has failures: true
Warning: /Stage[main]/Puppet_agent::Windows::Install/File[C:\Users\ADMINI~1\AppData\Local\Temp\install_puppet.bat]: Skip
ping because of failed dependencies
Notice: /Stage[main]/Puppet_agent::Windows::Install/Exec[install_puppet.bat]: Dependency File[C:\ProgramData\Puppetlabs\
packages\puppet-agent-x64.msi] has failures: true
Warning: /Stage[main]/Puppet_agent::Windows::Install/Exec[install_puppet.bat]: Skipping because of failed dependencies
Notice: /Stage[main]/Puppet_agent::Windows::Install/Exec[fix inheritable SYSTEM perms]: Dependency File[C:\ProgramData\P
uppetlabs\packages\puppet-agent-x64.msi] has failures: true
Warning: /Stage[main]/Puppet_agent::Windows::Install/Exec[fix inheritable SYSTEM perms]: Skipping because of failed dependencies
Notice: Finished catalog run in 2.30 seconds
```

I'm going to migrate this to a JIRA ticket

**Comment by MosesMendoza** [comment]
@Renelast filed as https://tickets.puppetlabs.com/browse/MODULES-4197

**Comment by MosesMendoza** [comment]
@Renelast I believe you might be running into an issue addressed by https://github.com/puppetlabs/puppetlabs-puppet_agent/commit/5876df98aa33c66f895e8c150417de41bc996702, released in puppet_agent 1.3.1 - if you run into this again, any chance you could that version a try?

**Comment by Renelast** [comment]
@MosesMendoza  That seems to address my issue indeed.
I'll update to the 1.3.1 release to test.
